pkg/endpoint/restore.go

```diff
@@ -61,36 +60,29 @@ func ReadEPsFromDirNames(ctx context.Context, owner regeneration.Owner, basePath
 	possibleEPs := map[uint16]*Endpoint{}
 	for _, epDirName := range completeEPDirNames {
 		epDir := filepath.Join(basePath, epDirName)
-		readDir := func() string {
-			scopedLog := log.WithFields(logrus.Fields{
-				logfields.EndpointID: epDirName,
-				logfields.Path:       filepath.Join(epDir, common.CHeaderFileName),
-			})
-			scopedLog.Debug("Reading directory")
-			epFiles, err := ioutil.ReadDir(epDir)
-			if err != nil {
-				scopedLog.WithError(err).Warn("Error while reading directory. Ignoring it...")
-				return ""
-			}
-			cHeaderFile := common.FindEPConfigCHeader(epDir, epFiles)
-			if cHeaderFile == "" {
-				return ""
-			}
-			return cHeaderFile
-		}
-		// There's an odd issue where the first read dir doesn't work.
-		cHeaderFile := readDir()
-		if cHeaderFile == "" {
-			cHeaderFile = readDir()
-		}
-
+		cHeaderFile := filepath.Join(epDir, common.CHeaderFileName)
 		scopedLog := log.WithFields(logrus.Fields{
 			logfields.EndpointID: epDirName,
 			logfields.Path:       cHeaderFile,
 		})
+		// This function checks the presence of a bug previously observed: the
+		// file was sometimes only found after the second check.
+		// We can remove this if we haven't seen the issue occur after a while.
+		headerFileExists := func() error {
+			var fileExists error
+			for i := 0; i < 3; i++ {
+				_, err := os.Stat(cHeaderFile)
+				if i > 0 && fileExists != err {
+					scopedLog.WithError(err).Warn("BUG: stat() has unstable behavior")
+				}
```

Comment:
Do we need to do the Stat if the file was found? I had the feeling that this only matters to be retried if the file is not found.